src/jobsautoreport/slack.py

```diff
@@ -14,87 +14,134 @@ def __init__(self, web_client: WebClient, channel_id: str) -> None:
         self._client = web_client
         self._channel_id = channel_id
 
-    def send_report(self, report: Report) -> None:
-        blocks = self._format_message(report)
-        logger.debug("Message format finished successfully")
+    def _post_message(
+        self,
+        report: Report,
+        format_function: Callable[[Report], list[dict[str, Any]]],
+        log_format_message: str,
```

Comment:
I think it's redundant to allow overriding the message after formatting. just do something generic, and you can understand which messages it's referring from by a log in the outer scope